public/app/core/services/backend_srv.ts

```diff
@@ -213,12 +213,12 @@ export class BackendSrv implements BackendService {
               const firstAttempt = i === 0 && options.retry === 0;
 
               if (error.status === 401 && isLocalUrl(options.url) && firstAttempt && isSignedIn) {
-                if (error.data.message.includes('revoked')) {
-                  appEvents.emit(CoreEvents.showModalReact, {
+                if (error.data?.message?.includes('revoked')) {
+                  this.dependencies.appEvents.emit(CoreEvents.showModalReact, {
```

Comment:
As you mentioned in the original PR, shouldn't we use the `appEvents.publish` method instead? 🤔

Reply:
I think this can wait until https://github.com/grafana/grafana/issues/31548 gets merged. I think I need to spend some time on refactoring the toFailureStream function anyways.

Reply:
I can't merge (have no rights) so go ahead and merge this.